docker/Dockerfile

```diff
@@ -0,0 +1,52 @@
+# A Docker image to run tests or experiment locally on a CPU machine
+# It should be based on /taskcluster/docker/test/Dockerfile
+
+ARG DOCKER_IMAGE_PARENT
+FROM $DOCKER_IMAGE_PARENT
+
+# Similar to /taskcluster/docker/toolchain/Dockerfile
+RUN apt-get update -qq \
+    # We need to install tzdata before all of the other packages. Otherwise it will show an interactive dialog that
+    # we cannot navigate while building the Docker image.
+    && apt-get install -y tzdata \
+    && apt-get install -y wget \
+                          curl \
+                          zip \
+                          build-essential \
+                          gcc \
+                          g++ \
+                          make \
+                          cmake \
+                          libboost-dev \
+                          libboost-all-dev \
+                          zstd \
+                          tar \
+                          libxml2 \
+                          libhunspell-dev \
+                          bc \
+    && apt-get clean
+
+RUN mkdir /builds/worker/tools && \
+    chown worker:worker /builds/worker/tools && \
+    mkdir /builds/worker/tools/bin && \
+    chown worker:worker /builds/worker/tools/bin
+
+WORKDIR /builds/worker/tools
+
+ADD pipeline/setup/* .
+
+ENV BIN=/builds/worker/tools/bin
+
+RUN git clone https://github.com/marian-nmt/extract-lex.git extract-lex
+RUN ./compile-extract-lex.sh extract-lex/build $(nproc)
+
+RUN git clone https://github.com/clab/fast_align.git fast_align
+RUN ./compile-fast-align.sh fast_align/build $(nproc)
+
+RUN git clone https://github.com/kpu/preprocess.git preprocess
+RUN ./compile-preprocess.sh preprocess/build $(nproc)
+
+RUN git clone https://github.com/marian-nmt/marian-dev.git marian-dev
+RUN ./compile-marian.sh marian-dev/build $(nproc) false
+
+ENV MARIAN=/builds/worker/tools/marian-dev/build
```

Comment:
Issue: I don't know why mine wouldn't work but your does, but I'm getting a python doesn't exist error. I found a solution though:

```diff
diff --git a/docker/Dockerfile b/docker/Dockerfile
index e9deacd..dc0aafa 100644
--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -50,3 +50,7 @@ RUN git clone https://github.com/marian-nmt/marian-dev.git marian-dev
 RUN ./compile-marian.sh marian-dev/build $(nproc) false
 
 ENV MARIAN=/builds/worker/tools/marian-dev/build
+
+# Work around poetry not finding `python`.
+# https://github.com/python-poetry/poetry/issues/6371
+RUN ln -s /bin/python3 /bin/python
```

Here is the error:

```
 => CACHED [10/13] RUN git clone https://github.com/kpu/preprocess.git preprocess                                                                                                                    0.0s
 => CACHED [11/13] RUN ./compile-preprocess.sh preprocess/build $(nproc)                                                                                                                             0.0s
 => CACHED [12/13] RUN git clone https://github.com/marian-nmt/marian-dev.git marian-dev                                                                                                             0.0s
 => CACHED [13/13] RUN ./compile-marian.sh marian-dev/build $(nproc) false                                                                                                                           0.0s
 => exporting to image                                                                                                                                                                               0.0s
 => => exporting layers                                                                                                                                                                              0.0s
 => => writing image sha256:72aff1a69e0b4667b695d2cdb07c3010e4006814908dd8be68a5bd9b7ff96f17                                                                                                         0.0s
 => => naming to docker.io/library/ftt-local                                                                                                                                                         0.0s

What's Next?
  View summary of image vulnerabilities and recommendations → docker scout quickview
if [ -n "$VIRTUAL_ENV" ]; then \
		echo "Error: Virtual environment detected. Exit the poetry shell."; \
		exit 1; \
	fi && \
	if [ $(uname -m) == 'arm64' ]; then \
		echo "setting arm64 platform"; \
	  	export DOCKER_DEFAULT_PLATFORM=linux/amd64; \
	fi && \
	docker run \
		--interactive \
		--tty \
		--rm \
		--volume $(pwd):/builds/worker/checkouts \
		--workdir /builds/worker/checkouts \
		 ftt-local make run-tests
setting arm64 platform
poetry install --only tests --only utils --no-root
PYTHONPATH=$(pwd) poetry run pytest tests -vv

[Errno 2] No such file or directory: 'python'

[Errno 2] No such file or directory: 'python'
make: *** [Makefile:57: run-tests] Error 1
make: *** [run-tests-docker] Error 2
```